### ISSUE TITLE
Incompatible enum type with base class

### ISSUE THREAD

**Issue by tefra** [tefra/xsdata#479]
```
(expression has type "Optional[IanaCountryTldEnumeration]", base class "EntityStructure" defined the type as "Optional[str]")
```

Generate base type for [enums](https://docs.python.org/3/library/enum.html#others)

**Comment by tefra** [comment]
This will be tricky to implement :disappointed:

**Comment by tefra** [comment]
It's way to tricky to handle these cases, the enum converter will need a lot of hacking and we will have a tone of issues with mixed type enumerations. I am thinking that for these cases we should ignore the subclass override type annotation.

**Comment by tefra** [comment]
closed in #516
